lighthouse-core/computed/metrics/cumulative-layout-shift.js

```diff
@@ -49,10 +53,22 @@ class CumulativeLayoutShift {
       throw new LHError(LHError.errors.LAYOUT_SHIFT_MISSING_DATA);
     }
 
+    // Chromium will set `had_recent_input` if there was recent user input, which
+    // skips shift events from contributing to CLS. This flag is also set when Lighthouse changes
+    // the emulation size. This consistently results in the first shift event always being ignored
+    // for CLS. Since we don't expect any user input, conditionally add the first shift event to
+    // CLS if it was ignored.
+    // See https://bugs.chromium.org/p/chromium/issues/detail?id=1094974.
+    if (firstLayoutShift.args.data && firstLayoutShift.args.data.had_recent_input) {
```

Comment:
shouldn't this be *all* events within 500ms of navStart?

Reply:
where does 500ms come from?

I was scoping just to the first to allow for some sort of scripted user interaction run of Lighthouse ...

Reply:
Also, I haven't ever seen a trace where the first two are marked as true.

Reply:
500ms comes from the LS API definition of marking layout shift events within 500ms of input as had recent input

See https://github.com/WICG/layout-instability#recent-input-exclusion

> Also, I haven't ever seen a trace where the first two are marked as true.

Just because our one example thus far doesn't have more than one doesn't mean more than one isn't possible ;)

https://melodic-class.glitch.me/cls-forever.html

Reply:
I had noticed this pattern when helping @Beytoven with `trace-elements` and saw only one for all the sites I happened to test.

confirmed your test case. +1

Reply:
![image](https://user-images.githubusercontent.com/4071474/87072002-b0b97280-c1e0-11ea-8c3a-02069116a393.png)

seems the time between nav start and the first shift event is larger than 500ms

let's just add the shift events back to CLS until `had_recent_input` is first false.

Reply:
> seems the time between nav start and the first shift event is larger than 500ms

I can consistently repro as well, but why is it happening?

The timer really is 500ms, and IIUC it shouldn't even be 500ms from navStart, it should be from when emulation was changed, which occurs on `about:blank`, before the navigation. Seems like we're still missing something that's happening here?

@anniesullie any thoughts?

Reply:
In my local build I added a trace event when the viewport resize occurs. I get:

navigationStart: 78ms
viewport resize: 1520ms
layout shift: 1668ms

So the resize event is received by the renderer much later than the navigationStart.

Reply:
Discussed in chat but the summary is that Chromium hangs on to viewport resize events from before a navigation and issues them to new renderers once they are created making a specific timeframe not possible to determine. 

Solution: if Chromium can decide to hang on to past resize events for arbitrary amount of time before issuing them to a renderer (and starting the 500ms countdown then) we should just ignore the flag until we see evidence that it's over. i.e. keep @connorjclark 's approach for now.